docs/src/main/asciidoc/maven-tooling.adoc

```diff
@@ -640,3 +631,82 @@ your default testing database.
+
+| `quarkus.bootstrap.effective-model-builder`
+| `false`
+| By default, the Quarkus Maven resolver is reading project's POMs directly when discovering the project's layout. While in most cases it works well enough and relatively fast, reading raw POMs has its limitation. E.g. if a POM includes modules in a profile, these modules will not be discovered. This system property enables project's layout discovery based on the effective POM models, that are properly interpolated, instead of the raw ones. The reason this option is not enabled by default is it may appear to be significantly more time consuming that could increase, e.g. CI testing times. Until there is a better approach found that could be used by default, projects that require it should enable this option.
+
+|===
+
+These system properties above could be added to, e.g., a `surefire` and/or `failsafe` plugin configuration as
+[source,xml,subs=attributes+]
+----
+<project>
+  [...]
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>${surefire-plugin.version}</version>
+        <configuration>
+          <systemPropertyVariables>
+            <maven.home>${maven.hom}</maven.home> <1>
+            <maven.repo.local>${settings.localRepository}</maven.repo.local> <2>
+            <maven.settings>${session.request.userSettingsFile.path}</maven.settings> <3>
+            <maven.top-level-basedir>${session.topLevelProject.basedir.absolutePath}</maven.top-level-basedir> <4>
+            <quarkus.bootstrap.effective-model-builder>true</quarkus.bootstrap.effective-model-builder> <5>
+          </systemPropertyVariables>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+  [...]
+</project>
+----
+
+<1> Propagate `maven.home` system property set by the Maven CLI to the tests
+<2> Set the Maven local repository directory for the tests
+<3> Set the Maven settings file the tests
+<4> Point to the top-level project directory for the tests
+<5> Enable effective POM-based project layout discovery
+
+==== Top-level vs Multi-module project directory
+
+In Maven there appears to be a notion of the top-level project (that is exposed as a project property `${session.topLevelProject.basedir.absolutePath}`)
+and the multi-module project directory (that is exposed as project property `${maven.multiModuleProjectDirectory}`). These directories do not always match!
+
+They would match in case the maven build was launched without the `-f` argument. If the `-f` argument is specified on the `mvn` command line though then
+the value of `${maven.topLevelProject.basedir.absolutePath}` will be pointing to the module that was targeted with the `-f` argument, while the value of
+`${maven.multiModuleProjectDirectory}` would be pointing to the root directory of the workspace.
```

Comment:
```suggestion
`${maven.multiModuleProjectDirectory}` would be pointing to the root directory of the workspace - but only if there is a top-level `.mvn` directory.
```
There could also be a link to http://takari.io/2015/03/20/mmp.html but that's obviously not the official maven site.

Reply:
Ok, I need to re-write it.

Reply:
How about the one I pushed?

Reply:
Looking good! 👍